### DISCUSSION TITLE
Cannot set custom font or directory

### DISCUSSION THREAD

**Discussion by amrishkakadiya** [dompdf/dompdf#2339] [General]
Hi, I am using Lumen 6.0

I am running my local server with taget public folder

![Directory](https://user-images.githubusercontent.com/37258496/103304469-bcb71c00-4a2e-11eb-99e8-cbad1647d64f.png)

here **RobotoSlab.afm** file manually pasted by me to see if it works.

I have created folder **fonts** in public folder 

and here is my code to generate pdf

![Code Snippet](https://user-images.githubusercontent.com/37258496/103304441-aa3ce280-4a2e-11eb-9e0f-cf40b76f3cb0.png)

But I am not getting my custom fonts in pdf. 
Generated PDF
![Generated PDF](https://user-images.githubusercontent.com/37258496/103304268-339fe500-4a2e-11eb-8c09-8384d64f6548.png)

Can you please help me, what is wrong with code or structure.
Thank you .

**Comment by bsweeney** [comment]
In order for Dompdf to use a font it has to be registered with Dompdf. This does a few things, like generate font metrics and update the font family cache. Have you read through the [Unicode How-To](https://github.com/dompdf/dompdf/wiki/UnicodeHowTo)? It has a good overview of the process, even though the page is more specifically about supporting non-ANSI character sets.

The easiest method is to use the `@font-face` CSS declaration. Something like:

```
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<link href="https://fonts.googleapis.com/css2?family=Roboto+Slab&display=swap" rel="stylesheet"> 
</head>
<body style="font-size: 20pt; font-family: 'Roboto Slab', sans-serif;">
This is a test
</body>
</html>
```

Other than that your settings look OK. The path specified for directory-based settings should be valid from the file calling Dompdf, so either a full path from the root of the drive or a path relative to that file.
